store/router.go

```diff
@@ -54,9 +54,21 @@ func (r *Router) Get(ctx context.Context, key []byte, cm commitments.CommitmentM
 			r.log.Debug("Retrieving data from memstore")
 			return r.mem.Get(ctx, key)
 		}
-
-		r.log.Debug("Retrieving data from eigenda")
-		return r.eigenda.Get(ctx, key)
+		if r.s3 != nil && r.s3.cfg.Backup {
+			r.log.Info("Retrieving data from S3", "key", crypto.Keccak256(key))
+			ctx2, cancel := context.WithTimeout(ctx, time.Minute)
+			value, err := r.s3.Get(ctx2, crypto.Keccak256(key))
+			cancel()
```

Comment:
we should probably defer the context 

```suggestion
			ctx2, cancel := context.WithTimeout(ctx, time.Minute)
			defer cancel()
			value, err := r.s3.Get(ctx2, crypto.Keccak256(key))
```